isofit/core/forward.py

```diff
@@ -93,11 +93,10 @@ def __init__(self, full_config: Config):
             self.surface = AdditiveGlintSurface(self.full_config)
         elif self.config.surface.surface_category == "glint_model_surface":
             self.surface = GlintModelSurface(self.full_config)
-            self.full_glint = (
-                self.config.surface.full_glint
-                if "full_glint" in self.config.surface.keys()
-                else False
-            )
+            try:
+                self.full_glint = self.config.surface.full_glint
```

Comment:
@evan-greenbrg Thanks for the comment! I don't think it gets more complicated when we move this block to the surface model class. The forward model has access to the surface model, so setting a new attribute there makes it automatically available to all classes and functions that utilize a forward model object. This includes the `invert_analytical` function.

Reply:
No problem at all! Apologies if this is pedantic. 

That's definitely a fair point that this flag could be a surface model attribute, and I think that makes a lot of sense. Another option is that if `full_glint` is included in the configs/sections/surface_config.py then `self.config.surface.full_glint` should already be set to either True or None and this line could be moved outside of the #545 block, and outside of the if-else statement entirely.

The other part of my question was related to not knowing exactly how `fm.RT.glint_model` and `fm.full_glint` differed. They are similar True-None glint-related flags and both used in inverse_simply.invert_analytical.

Reply:
Ah got it! I agree that both terms could be a bit confusing. Currently, `fm.RT.glint_model` simply indicates if the glint model should be used as surface model. `fm.full_glint` is a different flag that, if set to true, triggers the pipeline to output the full glint spectra instead of the two scaling factors:

```
g_dir = rho_ls * (t_down_dir / t_down_total)  # direct sky transmittance
g_dif = rho_ls * (t_down_dif / t_down_total)  # diffuse sky transmittance

if fm.full_glint:
    trajectory.append(trajectory[-1][-2] * g_dir)
    trajectory.append(trajectory[-1][-1] * g_dif)
```

The inversion trajectory initially only holds the two glint scaling factors at indices -1 and -2.

To your suggestion: Yes, I think that makes even more sense. `full_glint` should be a surface attribute and assigning it to the surface model class directly obtained from the configs/sections/surface_config.py without an if-else statement is more elegant.

Reply:
Makes sense! Thanks so much for clarifying.

Reply:
@evan-greenbrg I resolved a few merge conflicts and `full_glint` is now being set as an attribute to the surface class, obtained directly from the config file, following #545.